java-components/resource-model/src/main/resources/crds/jvmbuildservice.io_jbsconfigs.yaml

```diff
@@ -45,23 +45,13 @@ spec:
                 type: array
               buildSettings:
                 properties:
-                  buildRequestCPU:
```

Comment:
We can't do this for backward compatibility reasons (or at least I would need to make sure that these are not configured anywhere important). I think we should leave the ability to configure limits, but just leave it empty if unconfigured.

Reply:
Ok. Do you mean change the code around `resource.ParseQuantity(settingOrDefault(jbsConfig.Spec.BuildSettings.TaskRequestCPU, "10m"))` so that if it hasn't been configured don't configure it (and don't set a default) ? 

From reading further on this, I'm thinking we should actually set a requests value (perhaps to `1`) but no limits value.

Reply:
Hmm. Setting only requests gets
`Invalid request resource value: 500m must be less or equal to limit 0`
and the limit has been implicitly set to zero. Furthermore in the build step the build request value is copied over to the limits value:
```
          steps:
            - timeout: 6h0m0s
              resources:
                limits:
                  cpu: 300m
                  memory: 5Gi
                requests:
                  cpu: 300m
                  memory: 5Gi
              script: /original-content/build.sh "$@"
```

Reply:
Looks like it was our code taking the defaultRequestsLimit and making that the hard limit.

Reply:
@stuartwdouglas Should we add a separate ticket for `We can't do this for backward compatibility reasons (or at least I would need to make sure that these are not configured anywhere important). ` so we can deprecate and remove these?

Reply:
I don't think it is bad to leave this as configurable, there may be times when it is what we people want.

Reply:
Actually this PR means they now do nothing, so they should definitively be deprecated. I don't think it is worth worrying about wiring them up as an optional limit.